packages/playwright-core/src/client/network.ts

```diff
@@ -335,15 +336,24 @@ export class Route extends ChannelOwner<channels.RouteChannel> implements api.Ro
         body = Buffer.from(entry.body, 'base64');
     }
 
-    if (options.response) {
+    if (options.response instanceof APIResponse) {
       statusOption ??= options.response.status();
       headersOption ??= options.response.headers();
-      if (body === undefined && options.path === undefined && options.response instanceof APIResponse) {
+      if (body === undefined && options.path === undefined) {
         if (options.response._request._connection === this._connection)
           fetchResponseUid = (options.response as APIResponse)._fetchUid();
         else
           body = await options.response.body();
       }
+    } else if (options.response) {
+      const harResponse = options.response as HARResponse;
+      statusOption ??= harResponse.status;
+      headersOption ??= headersArrayToObject(harResponse.headers, false);
+      if (body === undefined && options.path === undefined) {
+        body = harResponse.content.text;
+        if (body !== undefined && harResponse.content.encoding === 'base64')
```

Comment:
`HARReponse.content.mimeType` is a required field - I think we should respect it here.